application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -701,6 +710,34 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                         } catch (Exception e) {
                         }
 
+                        PageLink pageLink = new PageLink(1000);
+                        PageData<User> users;
+                        do {
+                            List<ListenableFuture<?>> futures = new ArrayList<>();
+                            users = userService.findAllUsers(pageLink);
+                            for (User user : users.getData()) {
+                                futures.add(dbUpgradeExecutor.submit(() -> {
+                                    try {
+                                        log.info("Migrating password history for user: " + user.getId());
+                                        JsonNode additionalInfo = user.getAdditionalInfo();
+                                        if (additionalInfo != null && additionalInfo.has("userPasswordHistory")){
+                                            UserCredentials creds = userService.findUserCredentialsByUserId(user.getTenantId(), user.getId());
+                                            if (creds != null) {
+                                                creds.setAdditionalInfo(JacksonUtil.newObjectNode().set("userPasswordHistory", additionalInfo.get("userPasswordHistory")));
+                                                userService.saveUserCredentials(user.getTenantId(), creds);
+                                            }
+                                            ((ObjectNode) additionalInfo).remove("userPasswordHistory");
+                                            userService.saveUser(user);
+                                        }
+                                    } catch (Exception e){
+                                        log.error("Failed to migrate password history for user: " + user.getId(), e);
```

Comment:
log.error("Failed to migrate password history for user: [{}]", user.getId(), e);